scipy/stats/_entropy.py

```diff
@@ -163,40 +205,109 @@ def differential_entropy(
 
     """
     values = np.asarray(values)
-    values = np.moveaxis(values, axis, 0)
+    values = np.moveaxis(values, axis, -1)
+    n = values.shape[-1]  # number of observations
 
     if window_length is None:
-        window_length = math.floor(math.sqrt(len(values)) + 0.5)
+        window_length = math.floor(math.sqrt(n) + 0.5)
 
-    if not 2 <= 2 * window_length < len(values):
+    if not 2 <= 2 * window_length < n:
         raise ValueError(
             f"Window length ({window_length}) must be positive and less "
-            f"than half the sample size ({len(values)}).",
+            f"than half the sample size ({n}).",
         )
 
-    sorted_data = np.sort(values, axis=0)
+    if base is not None and base <= 0:
+        raise ValueError("`base` must be a positive number or `None`.")
 
-    repeats = np.array(
-        (window_length + 1,)
-        + ((1,) * (len(sorted_data) - 2))
-        + (window_length + 1,),
-    )
+    sorted_data = np.sort(values, axis=-1)
 
-    padded_data = np.repeat(
-        sorted_data,
-        repeats=repeats,
-        axis=0,
-    )
+    methods = {"vasicek": _vasicek_entropy,
+               "van es": _van_es_entropy,
+               "correa": _correa_entropy,
+               "ebrahimi": _ebrahimi_entropy,
+               "auto": None}
```

Comment:
I am not 100% sure this will work due to multiple lines.
This looks better than adding the type as I suggested I think. We would need to import `Dict` just for that otherwise.
```suggestion
               "auto": None}  # type: ignore
```